### PR TITLE
Product.locked is never updated

### PR DESCRIPTION
**Pull request by nguyenfilip**
ProductManager.applyProductChanges should update value of locked column for
existing product, based on the incoming product.

If (incorrectly) the value of locked isn't updated, then Products will be
always marked dirty even though they have never been changed (possible performance
problem)

### CODE REVIEW

**Comment by nguyenfilip** [comment]
@Ceiu can you please review this one?

**Comment by Ceiu** [comment]
ACK

File: server/src/main/java/org/candlepin/model/Product.java [hunk 1]
```diff
@@ -1290,10 +1290,6 @@ public boolean isChangedBy(ProductData dto) {
             return true;
         }
 
-        if (dto.isLocked() != null && !dto.isLocked().equals(this.locked)) {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/Product.java]: We still want this check in place

File: server/src/main/java/org/candlepin/controller/ProductManager.java [hunk 2]
```diff
@@ -511,6 +511,8 @@ private Product applyProductChanges(Product entity, ProductData update, Owner ow
             entity.setDependentProductIds(update.getDependentProductIds());
         }
 
+        entity.setLocked(update.isLocked());
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/ProductManager.java]: Just remembered, we need a null check on update.isLocked, in the event there's no change to the value.

```
if (update.isLocked() != null) {
    entity.setLocked(update.isLocked());
}
```

**Comment by nguyenfilip** [comment]
@Ceiu The inital plan to fix tests hasnt worked out. In the end I had to go with more hybrid approach. There are some tests (client_v1_size_spec) that legitimately require to modify product content in hosted mode throughout the test. So my initial plan to create content/product BEFORE the test starts wouldn't work here.

I kept a Trello card on a backlog to change this to more pure solution. The purest solution is to create/modify product/content in memory and have Candlepin pull it during refresh pools. But thats not how HostedTestSubscription* classes were initially designed.

**Pull request merged**